File: Visitor/Golang/DebitCard.go
package main

import "fmt"

type DebitCard struct {}

func (visitor *DebitCard) visitCPU(component *CPU)  {
	fmt.Println("Purchase CPU with Debit Card")
}

func (visitor *DebitCard) visitVGA(component *VGA)  {
	fmt.Println("Purchase VGA with Debit Card")
}

File: Visitor/Python 3/DebitCard.py
from IPayment import IPayment
from CPU import CPU
from VGA import VGA

class DebitCard(IPayment):
  def visit(self, component):
    if type(component) is CPU:
      print("Purchase CPU with Debit Card")
    elif type(component) is VGA:
      print("Purchase VGA with Debit Card")
